hack/pre-release.sh

```diff
@@ -148,6 +158,126 @@ update_OCP_version() {
   fi 
 }
 
+# This function handles updating the versions in GitHub
+# takes 3 arguments
+#
+# 1st arg: base branch to operate on 
+# 2nd arg: stream to update. Can be x, y or z to represent the x y or z stream.#
+# 3rd arg: current WMCO version
+github_update(){
+  if [ "$#" -ne 3 ]; then
+    echo incorrect parameter count for github_update $#
+    return 1
+  fi
+
+  base_branch=$1
+  stream=$2
+  wmco_version=$3
+
+  # check if the git branch is dirty, and exit if it is
+  check_git_branch
+
+  new_branch=$(create_git_branch_list $stream-stream-bump $base_branch)
+  switch_git_branch $base_branch $new_branch
+
+  # Locate and store the WMCO version from the makefile
+  # finds "WMCO_VERSION ?=" and selects text following it
+  update_WMCO_version $stream $wmco_version
+  update_OCP_version $stream $wmco_version
+
+  # check to see if the opeator-sdk version is correct
+  operator_sdk_current_version=$(operator-sdk version |  grep -oP 'operator-sdk version: "\K[^"]+' )
+  if [[ "$operator_sdk_current_version" != "$OPERATOR_SDK_VERSION" ]]; then
+    echo -e "${PINK}operator-sdk $OPERATOR_SDK_VERSION must be installed.${RESET}"
+    echo -e "${PINK}exiting.${RESET}"
+    git checkout $initial_branch
+    exit
+  fi
+
+  git submodule update --init --recursive
+  make bundle
+
+  commit_message="[$base_branch] Update version to $wmco_version
+
+  This commit was generated by hack/pre-release.sh"
+  generate_commit "$commit_message" build bundle Makefile config
+
+  # Display the newly created branches with push instructions
+  push_message=$(display_push_message "$new_branch")
+  echo
+  echo -e "${PINK}${push_message}${RESET}"
+  echo
+}
+
+# This function handles updating the versions in dist_git. Expects you to be connected to the VPN 
+# takes 2 arguments
+# 1st arg: stream to update. Can be x, y or z to represent the x y or z stream.#
+# 2nd arg: current WMCO version
+dist_git_update() { 
+  if [ "$#" -ne 2 ]; then
+    echo incorrect parameter count for dist_git_update $#
+    return 1
+  fi
+  stream=$1
+  wmco_version=$2 
+
+  # clones the dist-git repo into a temporary directory 
+  # this will be cleaned up at the end 
+  mkdir tmp
+  git clone https://pkgs.devel.redhat.com/git/containers/windows-machine-config-operator-bundle tmp/windows-machine-config-operator-bundle
+  DIST_GIT_LOCATION=tmp/windows-machine-config-operator-bundle
+
+  OCP_version=$(get_OCP_version $wmco_version)
+
+  echo "dist-git location set to $DIST_GIT_LOCATION"
+  current_dir=$PWD
+  cd $DIST_GIT_LOCATION
+
+  check_git_branch
+
+  # switch to the dist-git branch that corresponds with the current ocp version
+  # if a rhel-9 branch doesn't exist, switch to rhel-8
+  base_branch=rhaos-$OCP_version-rhel-9
```

Comment:
You should be able to authoritatively figure out if we need to use a rhel-9 or rhel-8 branch and not guess.

Reply:
should I make a function in common.sh for this? Or just list all of the branches and select the one with the right version?  Since there are some versions that have both rhel-8 and rhel-9 branches, I feel like I would have to pick between them anyway.

Reply:
I think we talked about removing dist-git branches that are stale as well, which could make this easier. Is there any reason we have both a rhel-8 and a rhel-9 branch for some versions if we only support one version of rhel on each version?

Reply:
It doesn't matter. Let us pick the branch authoritatively instead of depending on what is there in dist-git.